Tests/Algorithm/Framework/Portfolio/MinimumVariancePortfolioOptimizerTests.cs

```diff
@@ -0,0 +1,197 @@
+            var covariance3 = new double[,] { { 0.06, 0.09, 0.28 }, { 0.09, 0.25, 0.58 }, { 0.28, 0.58, 1.66 } };
+            var covariance4 = (double[,])null;
+
+            _historicalReturns = new List<double[,]>
+            {
+                historicalReturns1, 
+                historicalReturns2,
+                historicalReturns3,
+                historicalReturns4,
+                historicalReturns1,
+                historicalReturns2,
+                historicalReturns3,
+                historicalReturns4
+            };
+
+            _expectedReturns = new List<double[]>
+            {
+                expectedReturns1,
+                expectedReturns2,
+                expectedReturns3,
+                expectedReturns4,
+                expectedReturns1,
+                expectedReturns2,
+                expectedReturns3,
+                expectedReturns4
+            };
+
+            _covariances = new List<double[,]>
+            {
+                covariance1,
+                covariance2,
+                covariance3,
+                covariance4,
+                covariance1,
+                covariance2,
+                covariance3,
+                covariance4
+            };
+
+            _expectedResults = new List<double[]>
+            {
+                new double[] { -0.089212, 0.23431, -0.040975, 0.635503 },
+                new double[] { 0.366812, -0.139738, 0.49345 },
+                new double[] { 0.562216, 0.36747, -0.070314 },
+                new double[] { -0.119241, 0.443464, 0.437295 },
+                new double[] { -0.215505, 0.130699, 0.084806, 0.56899 },
+                new double[] { -0.275, 0.275, 0.45 },
+                new double[] { -0.129512, 0.551139, 0.319349 },
+                new double[] { 0.052859, 0.144177, 0.802964 },
+            };
+
+            _targetReturns = new Dictionary<int, double>
+            {
+                { 4, 0.15d },
+                { 5, 0.25d },
+                { 6, 0.5d },
+                { 7, 0.125d }
+            };
+        }
+
+        [TestCase(0)]
+        [TestCase(1)]
+        [TestCase(2)]
+        [TestCase(3)]
+        public void TestOptimizeWeightings(int testCaseNumber)
+        {
+            var testOptimizer = new MinimumVariancePortfolioOptimizer();
+
+            var result = testOptimizer.Optimize(
+                _historicalReturns[testCaseNumber],
+                _expectedReturns[testCaseNumber],
+                _covariances[testCaseNumber]);
+
+            Assert.AreEqual(_expectedResults[testCaseNumber], result.Select(x => Math.Round(x, 6)));
+            Assert.AreEqual(1d, result.Select(x => Math.Round(Math.Abs(x), 6)).Sum());
+        }
+
+        [TestCase(4)]
+        [TestCase(5)]
+        [TestCase(6)]
+        [TestCase(7)]
+        public void TestOptimizeWeightingsSpecifyingTargetReturns(int testCaseNumber)
```

Comment:
Let's remove `Test` from the test names, same comment for the other tests & files 👍